### PR TITLE
Data pooling

### LINKED ISSUE #109: Trainer.py doesn't use multiple workers for training when workers >1
`use_multiprocessing`  option needs to be `True` for model.fit_generator to initiate multiple generator instances 

Nice post demonstrating behaviors of model.fit_generator with and without multiprocessing:
https://keunwoochoi.wordpress.com/2017/08/24/tip-fit_generator-in-keras-how-to-parallelise-correctly/

Will fix this in the next PR

### PR DESCRIPTION
**Pull request by smguo**
This PR adds 2 main features: (1) data pooling (see #119) and (2) different data normalization options. Here are the major changes:

Data pooling is done using the CLI `dataset_pooling.py`. The CLI will copy source data to the destination directory, re-index the filenames, create new `frames_meta` file for the pooled dataset

Data normalization is done by computing the normalization parameters (median and inter-quartile range) in the preprocessing step. These parameters are only computed once and stored in the `frames_meta.csv` for easy access during tiling and inference. Median and inter-quartile range are used for normalization because they can scale all the input channels to similar ranges more robustly than mean and standard deviation, particularly in the case where different input channels have different non-Gaussian intensity distributions.

This PR also intends to fix #109 and #128.

Sorry for the delay in pushing this PR! Let me know if there is any question.

### CODE REVIEW

**Comment by smguo** [comment]
I fixed the testes locally and submitted the PR but then realized nosetests by default skipped the executables when running tests... So a bunch of tests didn't get run when I tested locally. I'll fix the rest of broken tests and comment here when the PR is ready for review

**Comment by jennyfolkesson** [comment]
@smguo I'm back from maternity leave now so just let me know when tests are fixed!

**Comment by jennyfolkesson** [comment]
@smguo woohoo I'm back in the biohub org - thanks! I'm busy tomorrow but I'll take a look on Wednesday.

**Comment by smguo** [comment]
@jennyfolkesson Welcome back! I made a few more updates to the inference and mask generation modules and need to update the docstrings (Sorry for the super long PR!). I'll let you know when the PR is ready for review.

Also, I think Biohub doesn't have Travis CI subscription anymore. Maybe we should set up Github action for testing first before reviewing and merging this PR? Let me know how you would like to proceed.

**Comment by jennyfolkesson** [comment]
Sounds good. I can make a separate PR with a github actions migration. Hopefully I can have it done soon - before you're done with your PR so you can merge with it before merging to master. Will let you know!

**Comment by smguo** [comment]
Sounds good @jennyfolkesson! Let me know. I'm happy to help set up github actions too if you're busy

**Comment by jennyfolkesson** [comment]
@smguo just finished the PR! :-)

**Comment by smguo** [comment]
Thanks! I'll take a look.

**Comment by smguo** [comment]
Merged `dl_mbl_2021` into this branch.

**Comment by mattersoflight** [comment]
This branch now contains changes to microDL codebase that were 
* used to produce Fig. 7 for the paper
* used during DL MBL 2021 course.
* enable robust virtual staining work during 2022. 
It makes sense to merge this branch and update the tests so that they comply with new features.
